typed-builder-macro/src/field_info.rs

```diff
@@ -190,10 +200,31 @@ impl<'a> FieldBuilderAttr<'a> {
                         } else {
                             return Err(Error::new_spanned(assign.right, "Expected string"));
                         }
-                        Ok(())
                     }
-                    _ => Err(Error::new_spanned(&assign, format!("Unknown parameter {:?}", name))),
+                    "requires" => {
+                        let Some((field_names, ordinal)) = field_names_with_current_orinal
+                        else {
+                            return Err(Error::new_spanned(assign, "`requires` is only allowed as a field attribute, not as a struct attribute."));
```

Comment:
I find this error message a bit confusing. I think something like "\`requires\` is not allowed in \`field_defaults\`" would better convey the meaning.